### PR TITLE
mysql doc rewrite

### PR DESCRIPTION
**Pull request by Hesperide**
Re-write of MySQL docs.

* Moves extra uncommonly needed info to a troubleshooting file.
* Matches format for Postgres and Mongo docs.
* Uses new tags `<HideInUI>`, `<details>` and `<FieldAnchor>`.
* Adds a handful of screenshots.

### CODE REVIEW

File: docs/integrations/sources/mysql.md
Comment (Hesperide) [inline docs/integrations/sources/mysql.md:10]:
```suggestion
![Airbyte MySQL Connection](https://raw.githubusercontent.com/airbytehq/airbyte/3a9264666b7b9b9d10ef8d174b8454a6c7e57560/docs/integrations/sources/mysql/assets/airbyte_mysql_source.png)
```

File: docs/integrations/sources/mysql.md
Comment (lmossman) [inline docs/integrations/sources/mysql.md:22]:
```suggestion
<FieldAnchor field="username,password">
```

I'm still investigating why this is the case, but the `#### Step 1:...` heading doesn't render properly in the Airbyte UI if there is an empty line between it and the <FieldAnchor> tag. 

I don't want to block this change on me investigating the bug so let's just get rid of the empty line here for now

**Review by lmossman — APPROVED** [review]
Left some suggestions but nothing truly blocking. LGTM once those are addressed

**Review by akashkulk — APPROVED** [review]
Nice! Is the troubleshooting.md file linked from anywhere? Couldn't find the link to it from the main mysql set up page.

File: docs/integrations/sources/mysql.md
Comment (Hesperide) [reply docs/integrations/sources/mysql.md:10]: Common practice to include a screenshot of what something looks like when you are done at the beginning of instructions :)

File: docs/integrations/sources/mysql.md
Comment (Hesperide) [reply docs/integrations/sources/mysql.md:213]: I would advocate for this, but would prefer to do this all at once for all connectors.

File: docs/integrations/sources/mysql.md
Comment (Hesperide) [reply docs/integrations/sources/mysql.md:82]: I see where you are coming from (and wouldn't be strongly opposed), but we also have a step 4. And I don't want us to have to worry about Step 1 -> 2 -> 4 (skipping 3).

Maybe something we can look at improving in the future.

**Pull request merged**
